cmd/gen-manifests/main.go

```diff
@@ -161,7 +161,13 @@ func makeManifestJob(name string, imgType distro.ImageType, cr composeRequest, d
 			return fmt.Errorf("[%s] container resolution failed: %s", filename, err.Error())
 		}
 
-		packageSpecs, err := depsolve(cacheDir, imgType, bp, options, repos, distribution, archName)
+		manifest, _, err := imgType.Manifest(&bp, options, repos, nil, containerSpecs, seedArg)
+		if err != nil {
+			err = fmt.Errorf("[%s] failed: %s", filename, err)
+			return
+		}
+
+		packageSpecs, err := depsolve(cacheDir, manifest.Content.PackageSets, distribution, archName)
```

Comment:
Seeing `manifest.Content.PackageSets`, did you consider adding methods to the `Manifest` struct for getting its content? I'm OK with this, but since we tend to use interfaces a lot, I wanted to ask...

Reply:
The funny thing is that these methods exist and the do the collection from the pipelines.  I made the `Content` field public so that it can be set externally for testing, thinking it might be useful for other purposes as well, but it turned out not to be.  I think I'd rather have the content be private and only accessible through the methods, and the methods can be written to do the collection only once to avoid scanning through every time.  I'm trying out something to get around the testing.

I could turn Manifest into an interface as well, but I'll see if I can avoid that.

Reply:
So here's where I'm at:
- I don't want to make the `manifest.Manifest` an interface.
- We have the manifest.Pipeline interface, but it has private methods so you can't override them outside the manifest package.
- I'd like to be able to make simple `Manifest` instances that return fixed content and maybe fixed pipelines for testing.
- Figuring out how to test these things helps us figure out how users who import the packages might interact with them.

What I'm thinking now is to keep `Manifest` as a struct, make `AddPipeline()` public, but simplify the `manifest.Pipeline` interface and make some of its methods public, while keeping other private methods implemented in the `Base`.  This way anyone from outside the `manifest` package can implement custom pipelines by embedding the `Base` pipeline.  This is already possible now, but too many of the private methods in `Base` simply return `nil` and aren't useful.

The basic question is: do we want external users writing their own pipelines?  I think we do.
I think the `Manifest` is simple enough that it doesn't need to be an interface.  The collection and serialization methods should have clear semantics (and I believe they already do) so that it's straightforward to implement one or more custom Pipelines to get a desired output.

Reply:
Still fiddling with this.  I think I might leave it as is and go deeper in a follow-up.

Reply:
I'm going to keep the `Content` field public on `Manifest` for now, solely in service of keeping `test_distro` mostly unchanged.  I think I will immediately follow this PR up with another one that makes the content private and handles the testing differently.

Reply:
Your plan sounds good to me. 👍